### DISCUSSION TITLE
Deprecate the Kernel Extension

### DISCUSSION THREAD

**Discussion by tnek** [google/santa#671] [General]
We're approaching 3 major versions since kexts have been deprecated. Deprecating the santa driver also cleans up all of the install scripts and makes maintaining/changing things in Common easier

**Comment by russellhancox** [comment]
I'd like for us to give time to users to provide feedback before we go ahead with this. While it's quite likely there are no remaining users of the kernel extension, not every organization upgrades macOS as promptly as {we do, they should} for various reasons. 

We can post an announcement and RFC on the santa-dev group, a discussion on GitHub and possibly a notice to the MacAdmins slack #santa channel

**Comment by arubdesu** [comment]
I see the persuasive case being made there's process improvements that would come from being able to focus on the one officially supported low-level hook mechanism, since it currently isn't a choice Santa allows you to make on a recent OS. Is this just as simple as telling people on old OS versions to please continue using the releases prior to a specific tag? Munki has found that an easy-enough way to remove support for older OS versions without too much disruption that I can recall noticing.

**Comment by pmarkowsky** [comment]
I've made the following Google Form for folks to tell us if they're using the Kernel Extension https://forms.gle/adR6K5yvLcDc3ZWD7
